treepoem/__main__.py

```diff
@@ -0,0 +1,35 @@
+# -*- encoding:utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import argparse
+import sys
+from textwrap import fill
+
+from . import generate_barcode
+from .data import barcode_types
+
+supported_barcode_types = (
+    'Supported barcode types are:\n'
+    + fill(', '.join(sorted(barcode_types)), initial_indent='    ', subsequent_indent='    ')
+)
+
+
+def main():
+    p = argparse.ArgumentParser(epilog=supported_barcode_types)
```

Comment:
rename to `parser`, it's nicest to use >1 char for var names that aren't temporary

Also since argparse is declarative I find it's nicer to define the parser at module level, outside of the function that's using it, so the actual function is smaller and easier to understand